src/settings/toml/builder.rs

```diff
@@ -0,0 +1,66 @@
+use std::env;
```

Comment:
The new additions would look something like this for a durable object

```
[builder_config]
build_dir = "./build"
src_dir = "./"
upload_format = "modules"

[durable_objects]
implements = [
  { class_name = "Counter" }
]
```

or for a regular service worker just

```
[builder_config]
build_command = "./dummy-build.sh"
build_dir = "./my-build-dir"
src_dir = "./does-not-exist"
upload_format = "service-worker"
```

Reply:
Lots of underscores and duplicate info... Perhaps it could be streamlined?
```toml
[build]
cwd = "./build"
command = "sh ./dummy-build.sh"
src = "."
format = "modules"

[durable_objects] # can't really avoid that one
implements = [{ class = "Counter" }]
```

I say this because toml is pretty flexible with how you can write out tables and while you may write a table like that, I may write it like
```toml
build.cwd = "./build"
build.command = "sh ./dummy-build.sh"
build.src = "."
build.format = "modules"

durable_objects.implements = [{ class = "Counter" }]
```
With the long names you get
```toml
builder_config.build_dir = "./build"
builder_config.build_command = "sh ./dummy-build.sh"
builder_config.src_dir = "."
builder_config.upload_format = "modules"

durable_objects.implements = [{ class_name = "Counter" }]
```
Toml accepts both but shorter names look better with the second format.

Reply:
I tend to prefer verbosity when it eliminates confusion, but I agree on `build_command` -> `command`.
I'll defer to @greg-mckeon on the rest

Reply:
I like `command`, and I think that `src_dir` should be `working_dir`. And I mentioned earlier that `build_dir` should be `upload_dir`.

Reply:
+1 for command, src_dir does make sense to me if it's where you're building from.  If build_dir is where we're storing the output, I think it should be called target_dir to match, but open to upload_dir if you think that makes sense.

Could we change builder_config to just be build? That would also match with durable_objects.

Reply:
> I like `command`, and I think that `src_dir` should be `working_dir`. And I mentioned earlier that `build_dir` should be `upload_dir`.

currently, src_dir is only used for commands that watch for changes like live preview and wrangler dev, it's not used to set the working directory of the build command. The working directory of the build command should currently just be wherever the user decides to invoke "wrangler", which will probably be the project root as that's where `wrangler.toml` would be.

Reply:
> currently, src_dir is only used for commands that watch for changes like live preview and wrangler dev, it's not used to set the working directory of the build command.

That's very confusing. If it's only used for watching, why not call it `watch` or `watch_dir`?

Reply:
> > currently, src_dir is only used for commands that watch for changes like live preview and wrangler dev, it's not used to set the working directory of the build command.
> 
> That's very confusing. If it's only used for watching, why not call it `watch` or `watch_dir`?
if I did that rename, would you want a separate option for configuring the current working directory?

Reply:
I'd personally want it to be configurable with reasonable defaults. So you can choose a different directory to watch, but it uses the build command's cwd by default, or perhaps just the root folder (where wrangler.toml is).

Reply:
Why are we talking about configuring cwd?
Wrangler won't work anywhere but the directory with wrangler.toml
If someone needs a different cwd for their build, they can change to it and back in their build script.

Reply:
Thanks for explaining @xortive. I think `src_dir` is good.

Reply:
ah :/ I made the change to watch_dir already. It's easy to put it back, and I'm completely indifferent to which one we pick

Reply:
Oh that's fine idc either way.